protobuf/flagd/sync/v1/schema.proto

```diff
@@ -0,0 +1,93 @@
+/**
+ * Flag definition sync API
+ *
+ * This proto defines a simple API to synchronize a feature flag definition.
+ * It supports establishing a stream for getting notifications about changes in a flag definition.
+ */
+syntax = "proto3";
+
+package flagd.sync.v1;
+
+option csharp_namespace = "OpenFeature.Flagd.Grpc.Sync";
+option go_package = "flagd/sync/v1";
+option java_package = "dev.openfeature.flagd.grpc.sync";
+option php_namespace = "OpenFeature\\Providers\\Flagd\\Schema\\Grpc\\Sync";
+option ruby_package = "OpenFeature::FlagD::Provider::Grpc::Sync";
+
+// SyncFlagsRequest is the request initiating the sever-streaming rpc. Flagd sends this request, acting as the client
+message SyncFlagsRequest {
+  // Optional: A unique identifier for flagd(grpc client) initiating the request. The server implementations may
+  // utilize this identifier to uniquely identify, validate(ex:- enforce authentication/authorization) and filter
+  // flag configurations that it can expose to this request. This field is intended to be optional. However server
+  // implementations may enforce it.
+  //    ex:- provider_id: flagd-weatherapp-sidecar
+  string provider_id = 1;
+
+  // Optional: A selector for the flag configuration request. The server implementation may utilize this to select
+  // flag configurations from a collection, select the source of the flag or combine this to any desired underlying
+  // filtering mechanism.
+  //    ex:- selector: 'source=database,app=weatherapp'
+  string selector = 2;
+}
+
+// SyncState conveys the state of the payload. These states are related to flagd isync.go type definitions but
+// contains extras to optimize grpc use case. Refer - https://github.com/open-feature/flagd/blob/main/pkg/sync/isync.go
+enum SyncState {
+  // Value is ignored by the listening flagd
+  SYNC_STATE_UNSPECIFIED = 0;
+
+  // All the flags matching the request. This is the default response and other states can be ignored
+  // by the implementation. Flagd internally replaces all existing flags for this response state.
+  SYNC_STATE_ALL = 1;
+
+  // Convey an addition of a flag. Flagd internally handles this by combining new flags with existing ones
+  SYNC_STATE_ADD = 2;
+
+  // Convey an update of a flag. Flagd internally attempts to update if the updated flag already exist OR if it does not,
+  // it will get added
+  SYNC_STATE_UPDATE = 3;
+
+  // Convey a deletion of a flag. Flagd internally removes the flag
+  SYNC_STATE_DELETE = 4;
```

Comment:
I think this is a good time to discuss the usefulness of these states. 

When we initially introduced sync states, the intention was to provide atomic updates. However, the server-side implementation of such an update mechanism is complex and requires state maintenance. 

Given that we are migrating to a newer schema, I think this is the right time to do modifications. 

Do we need to keep these states ? Or shall we remove them and only keep SYNC ALL ?

cc - @thisthat @toddbaert @beeme1mr @bacherfl  opinions are welcome on this

Reply:
The big benefit I can see is that if we have huge sets of flags the diffing could allow much smaller payloads. This would be especially important if there was huge sets of flags which were frequently changed. However, you're right that implementing this in any server adds complexity, and more statefulness.

The `selector` is also relevant here, since it should identify a set of relevant flags - a org might have 10K flags defined in it's system... but we would not expect all of those to be in the same scope, I think... hopefully reducing the chance of massive payloads.

Reply:
We could also reintroduce this in a non-breaking way if performance ever becomes an issue.

Reply:
I _think_ we can completely remove this for now. I can't think of much of a benefit to keeping only the `SYNC_ALL`. That might be a good choice if adding this field later would be a breaking change, but I don't think think it will be. Since it's not being used and adding it should be non-breaking, I believe fully removing it to be the best choice for now.

Any counterpoints?

Reply:
This is a breaking change and given that we are moving to a new naming (changing `package`) we can completely remove `SyncState`. When we see the need (or if there's a community request to add the field), then we can add the enum back. 

In gRPC, adding a field to a request is a non-breaking change. So going ahead with the least required fields with this change is the best option.

Reply:
@toddbaert  sorry, I should have been more specific on my last remark

I wanted to refer the logic bound to flagd `iSync` [^1]. Here as you can see, we have internal enum mapping to support this grpc update (All, Add, update, delete..). We can let this logic live as this is stable and well-tested. 

But we can remove the enum completely at grpc definition for the next version. And introduce it if we see the need. 


[^1]: https://github.com/open-feature/flagd/blob/main/core/pkg/sync/isync.go

Reply:
ahh, got it.

Reply:
OK let's remove this enum then! @bacherfl

Reply:
> I agree with the removal then.

 yes I also agree on the removal of enum from grpc definition. All flagd systems that I know do not use this right now.

Reply:
alright, i have removed the enum, so the PR should be ready for another review :)